contracts/services/astroport-lper/src/msg.rs

```diff
@@ -0,0 +1,164 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{ensure, Addr, Decimal, Deps, Uint128};
+use cw_ownable::{cw_ownable_execute, cw_ownable_query};
+
+use crate::error::ServiceError;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub owner: String,
+    pub processor: String,
+    pub config: ServiceConfig,
+}
+
+pub trait ServiceConfigValidation<T> {
+    fn validate(&self, deps: Deps) -> Result<T, ServiceError>;
+}
+
+#[cw_ownable_execute]
+#[cw_serde]
+pub enum ExecuteMsg {
+    ProcessAction(ActionsMsgs),
+    UpdateConfig { new_config: ServiceConfig },
+    UpdateProcessor { processor: String },
+}
+
+#[cw_serde]
+pub enum ActionsMsgs {
+    ProvideDoubleSidedLiquidity {
+        expected_pool_ratio_range: Option<DecimalRange>,
+    },
+    ProvideSingleSidedLiquidity {
+        asset: String,
+        limit: Option<Uint128>,
+        expected_pool_ratio_range: Option<DecimalRange>,
+    },
+}
+
+#[cw_serde]
+pub struct DecimalRange {
+    min: Decimal,
+    max: Decimal,
+}
+
+impl DecimalRange {
+    pub fn is_within_range(&self, value: Decimal) -> Result<(), ServiceError> {
+        ensure!(
+            value >= self.min && value <= self.max,
+            ServiceError::ExecutionError("Value is not within the expected range".to_string())
+        );
+        Ok(())
+    }
+}
+
+#[cw_ownable_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(Addr)]
+    GetProcessor {},
+    #[returns(Config)]
+    GetServiceConfig {},
+}
+
+#[cw_serde]
+pub struct ServiceConfig {
+    pub input_addr: String,
+    pub output_addr: String,
+    pub pool_addr: String,
+    pub lp_config: LiquidityProviderConfig,
+}
+
+#[cw_serde]
+pub struct LiquidityProviderConfig {
```

Comment:
Something that is not in the spec, but was discussed afterwards: it would be nice to expose any setting from the underlying pool in the config (or LPing messages) e.g. Astroport has an `auto_stake` option.

Reply:
I don't think we want to give the option to stake the LP tokens, If we want an astroport staker service we should implement that separately